pkg/cortex/modules.go

```diff
@@ -801,18 +802,18 @@ func (t *Cortex) initChunksPurger() (services.Service, error) {
 	return t.Purger, nil
 }
 
-func (t *Cortex) initTenantDeletionAPI() (services.Service, error) {
+func (t *Cortex) initBlocksPurger() (services.Service, error) {
 	if t.Cfg.Storage.Engine != storage.StorageEngineBlocks {
 		return nil, nil
 	}
 
 	// t.RulerStorage can be nil when running in single-binary mode, and rule storage is not configured.
-	tenantDeletionAPI, err := purger.NewTenantDeletionAPI(t.Cfg.BlocksStorage, t.Overrides, util_log.Logger, prometheus.DefaultRegisterer)
+	blockPurger, err := purger.NewBlocksPurgerAPI(t.Cfg.BlocksStorage, t.Overrides, util_log.Logger, prometheus.DefaultRegisterer, t.Cfg.PurgerConfig.DeleteRequestCancelPeriod)
 	if err != nil {
 		return nil, err
 	}
 
-	t.API.RegisterTenantDeletion(tenantDeletionAPI)
+	t.API.RegisterBlocksPurger(blockPurger, t.Cfg.PurgerConfig.Enable)
```

Comment:
This flag could be renamed to a more friendly name: like `EnableBlocksSeriesDeletion`?

Reply:
I've renamed the variable to `EnableSeriesDeletion`. I didn't change the flag name itself since the flag can be shared among both the chunks and blocks implementation. The current description of the flag is: `Enable purger to allow deletion of series. Be aware that Delete series feature is still experimental`.